include/circt/Dialect/FIRRTL/FIRRTLFieldSource.h

```diff
@@ -28,50 +28,20 @@ class FieldSource {
 
 public:
   explicit FieldSource(Operation *operation);
-
+  
   struct PathNode {
-    PathNode(Value src, ArrayRef<int64_t> ar, Flow flow)
-        : src(src), flow(flow), path(ar) {}
     Value src;
-    Flow flow;
-    SmallVector<int64_t, 4> path;
-
-    /// Roots are operations which define the storage or aggregate value.
-    bool isRoot() const { return path.empty(); }
-
-    /// Writable sources can appear as a LHS of a connect given this node's
-    /// path.
-    bool isSrcWritable() const { return flow != Flow::Source; }
-
-    /// Transparent sources reflect a value written to them in the same cycle it
-    /// is written.  These are sources which provide dataflow backwards in SSA
-    /// during one logical execution of a module body.
-    /// A port may be transparent (no storage), but not be writable from the
-    /// side of the instance we are on.
-    bool isSrcTransparent() const {
-      // ports are wires
-      if (!src.getDefiningOp())
-        return true;
-      // ports are wires, on this side of the instance too.
-      return isa<WireOp, InstanceOp>(src.getDefiningOp());
-    }
+//    Flow flow;
+    size_t fieldID;
```

Comment:
This is no longer a PathNode, seems like it's a FieldRef -- structurally at least.

Please rename it and update relevant code/comments to reflect what this analysis and its datastructures are/do now.